tests/unittests/core/fileio/fileutilstest.cpp

```diff
@@ -0,0 +1,313 @@
+/*
+ * LibrePCB - Professional EDA for everyone!
+ * Copyright (C) 2013 LibrePCB Developers, see AUTHORS.md for contributors.
+ * https://librepcb.org/
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+/*******************************************************************************
+ *  Includes
+ ******************************************************************************/
+
+#include <gtest/gtest.h>
+#include <librepcb/core/exceptions.h>
+#include <librepcb/core/fileio/filepath.h>
+#include <librepcb/core/fileio/fileutils.h>
+#if defined(Q_OS_WIN32) || defined(Q_OS_WIN64)  // Windows
+#include <fileapi.h>
+#endif
+
+#include <QtCore>
+
+/*******************************************************************************
+ *  Namespace
+ ******************************************************************************/
+namespace librepcb {
+namespace tests {
+
+/*******************************************************************************
+ *  Test Class
+ ******************************************************************************/
+
+class FileUtilsTest : public ::testing::Test {
+protected:
+  FilePath root;
+  FilePath rootFile;  // source for all operations
+  FilePath rootFileHidden;
+
+  FilePath subdir;
+  FilePath subdirFile;
+  FilePath subdirSubdir;
+  FilePath subdirSubdirFile;
+  FilePath subdirSubdirFileHidden;
+
+  FilePath rootFileMissing;
+  FilePath rootFileCopy;
+  FilePath subdirCopy;
+  FilePath subdirCopyFile;
+  FilePath subdirCopySubdir;
+  FilePath subdirCopySubdirFile;
+  FilePath subdirCopySubdirFileHidden;
+
+  QStringList filter;
+
+  static void SetupFile(const FilePath& pth, const QByteArray& content) {
+    QFile f(pth.toNative());
+    if (f.open(QIODevice::ReadWrite | QIODevice::NewOnly)) f.write(content);
+    f.close();
+  }
+
+  void SetUp() override {
+    root = FilePath::getRandomTempPath();
+
+    // the sources of already existing files and directories
+    rootFile = FilePath::fromRelative(root, "file.txt");
```

Comment:
Just FYI but nothing critical no no need to change here: Basically everywhere else we use this form (it is also shorter):

```suggestion
    rootFile = root.getPathTo("file.txt");
```

Reply:
Yeah, I like the other norm, I will corect the code to use that way. 

I am still new in codebase, looking what and how it is made in other tests :-).